### PR TITLE
Enable document generation using rosdoc2 for ament_python pkgs

### PR DESCRIPTION
**Pull request by Yadunund**
Fix missing `exec_depends` in `package.xmls` and docstrings to generate documentation without any warnings using `autodoc`

### CODE REVIEW

**Review by MichaelOrlov — APPROVED** [review]
@Yadunund LGTM.
I am curious if we need to add the same dependencies to the https://github.com/ros2/rosbag2/tree/rolling/rosbag2_py package?

**Review by clalancette — COMMENTED** [review]
These are obviously correct, so that's good.

While we are in here, though, we should probably clean up the other depends right above.  In particular, both of `ros2cli` and `rosbag2_py` should be `exec_depend`, not `depend`.  And this package doesn't directly depend on `rosbag2_transport` at all, so that should be removed.

**Comment by Yadunund** [comment]
Thanks @clalancette for pointing out correction to the dependencies!

@MichaelOrlov I think `sphinx` can generate documentation for pybind11 bindings but we'll need to add some custom `rosdoc2.yaml` and `conf.py` files. I'm currently focussing on `ament_python` pkgs and can take a look at `rosbag2_py` in the future.

**Review by clalancette — APPROVED** [review]
This looks good to me now, thanks.

I'm going to run CI on it next, but I'll wait for another approval from @MichaelOrlov before merging.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/18214/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/12745/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/18883/)

**Review by MichaelOrlov — APPROVED** [review]
LGTM.

**Comment by MichaelOrlov** [comment]
https://github.com/Mergifyio backport humble

**Pull request merged**
